web/src/components/MemoContent/Link.tsx

```diff
@@ -1,18 +1,98 @@
+import { Tooltip, Card, AspectRatio, Box } from "@mui/joy";
+import { Link as MLink } from "@mui/joy";
+import { useEffect, useState } from "react";
+import { linkServiceClient } from "@/grpcweb";
+import useResponsiveWidth from "@/hooks/useResponsiveWidth";
+import { LinkMetadata } from "@/types/proto/api/v2/link_service";
+
 interface Props {
   url: string;
   text?: string;
 }
 
 const Link: React.FC<Props> = ({ text, url }: Props) => {
+  const [linkMetadata, setLinkMetadata] = useState<LinkMetadata | undefined>();
+  const { sm, md } = useResponsiveWidth();
+
+  const fetchUrlMetadata = async () => {
+    try {
+      const response = await linkServiceClient.getLinkMetadata({ link: url }, {});
+      setLinkMetadata(response.metadata);
+    } catch (error) {
+      console.error("Error fetching URL metadata:", error);
+      return null;
+    }
+  };
+
+  useEffect(() => {
+    fetchUrlMetadata();
+  }, [url]);
+
   return (
-    <a
-      className="text-blue-600 dark:text-blue-400 cursor-pointer underline break-all hover:opacity-80 decoration-1"
-      href={url}
-      target="_blank"
-      rel="noopener noreferrer"
-    >
-      {text || url}
-    </a>
+    <>
+      {/* Small screens */}
+      {sm && (
+        <div className="flex md:hidden">
+          <MLink href={url} underline="none" sx={{ fontWeight: "lg" }}>
+            {url || text}
+          </MLink>
+        </div>
+      )}
+
+      {/* Medium & Above screens */}
+      {md && (
```

Comment:
```
{
  md ? <>Card block</> : <>Link block</>
}
```

Reply:
How about we remove the check for `sm` display and instead only display Preview Card on medium and above screens and show a link on small screens like you suggested and completely removing the `sm` block

```
{
  md ? <>Card block</> : <>Link block</>
}
```